### ISSUE TITLE
Versions 3.5.0 and 3.6.0 not picking .flake8 configuration file

### ISSUE THREAD

**Issue by asottile** [PyCQA/flake8#665]
In GitLab by @rafa.munoz on Jan 25, 2019, 04:19

System details (`python:2.7` Docker image):

```
Python 2.7.15 (default, Nov 16 2018, 23:13:11)
[GCC 6.3.0 20170516] on linux2
```

I set my `.flake8` file in the parent directory:

```
[flake8]
exclude = settings,manage.py,wsgi.py,migrations
```

If I try `pip install flake8==3.5.0` or `pip install flake8==3.6.0`, when running `flake8 .` in the parent directory, it will ignore the configuration file and will throw many errors. If I install version `3.4.0` it will read the config file and it will finish successfully.

Strangely, this started to happen this week in our builds. Before, we were working well with version `3.5.0`.

**Comment by asottile** [comment]
In GitLab by @patallen on Jan 25, 2019, 04:24

Looks like it was a regression in the 3.5.2 release of `configparser`. They've created a patch, but obviously is not released. Locking `configparser==3.5.0` in our requirements file fixed this for us. I assume locking to 3.5.1 will work as well.

Not sure why flake8 does not lock the version for `configparser` at all in `setup.py`, but this would fix this issue if locked to a minor version.

**Comment by asottile** [comment]
In GitLab by @rafa.munoz on Jan 25, 2019, 04:36

Thank you, Patrick, for the fast response.

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Jan 25, 2019, 04:37

@patallen This is the first issue I've seen with `configparser`, so I'll definitely take it under advisement to restrict it going forward. To be fair, though, we only use bounds and based on the bounds we use, 3.5.2 would have been acceptable.

**Comment by asottile** [comment]
In GitLab by @patallen on Jan 25, 2019, 04:42

@sigmavirus24 Absolutely. Locking it the way you do normally wouldn't fix it, and you shouldn't be expected to lock to patch versions IMO. No blame on you, just an observation.

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Jan 25, 2019, 04:50

Thanks for clarifying @patallen. All too often people comment "Not sure why ${project} does (not)? do ${X}" is uttered in rage, disgust, and/or a variety of other negative emotions that only serve to beat down the motivation of maintainers. Glad that wasn't the case here.

**Comment by asottile** [comment]
In GitLab by @clokep on Jan 25, 2019, 06:53

configparser 3.5.3 was released which fixes this problem for me.

Unfortunately configparser 3.7.0 was also released which breaks flake8 for me on Python 2.7.15 (see https://github.com/jaraco/configparser/issues/30).

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Jan 26, 2019, 05:37

@clokep Seems like 3.7.0 also had a minor release to fix that issue.

**Comment by asottile** [comment]
In GitLab by @asottile on Jan 29, 2019, 20:00

Upstream provided fixes for both the 3.5.x and 3.7.x releases.  Apologies since it was actually my patch which broke this I think :grimacing:

**Comment by asottile** [comment]
In GitLab by @asottile on Jan 29, 2019, 20:00

closed
